### DISCUSSION TITLE
[ Question ] Reverse proxy domain path to cloud function or AWS Lambda service?

### DISCUSSION THREAD

**Discussion by Borillion** [evertramos/nginx-proxy-automation#375] [Q&A] [answered]
I would like to proxy a location on my host https://someurl.com/consumer ->  https://project_id.cloudfunctions.net

In a non-docker setup I would write something like so in a someurl_com.conf file:
    
    server {
        listen 443 ssl;
        server_name someurl'.com;
    
        ssl_certificate /path/to/certificate;
        ssl_certificate_key /path/to/key;
    
        location /consumer {
            proxy_pass https://project_id.cloudfunctions.net/;
            proxy_set_header Host $host;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
            underscores_in_headers on;
            proxy_set_header proxy_ssl_server_name $host;
        }
    }

I am trying to figure out what I would put in the compose.yml file to cause the proper configuration to be generated by nginx.tmpl but things get weird, docker requires each service to have an image. 

Is there best practice method to work around this?

**Comment by evertramos** [comment, marked as answer]
Hello @Borillion,

Thanks for your message.

Let me see if I got this right. You have the scenario below:

### Container 1 with: 
- "-e VIRTUAL_HOST=someurl.com"

### Container 2 with: 
- "-e VIRTUAL_HOST=project_id.cloudfunctions.net"

And you want the container 1 to forward the location '/consumer' to container 2 as of:

https://someurl.com/consumer -> https://project_id.cloudfunctions.net/

If you would be using microservices and have one container to respond for 'someurl.com/consumer ' by itself (not two domains), 
you could use [path based routing](https://github.com/nginx-proxy/nginx-proxy#path-based-routing) which works perfectly for that. But in this case might mix the 'VIRTUAL_PATH' as you have two domains, did not really tested this... 

I would suggest you to override the location as of:

Place a files called 'someurl.com_location_override' to the 'vhost.d' folder, which should match the ['NGINX_FILES_PATH](https://github.com/evertramos/nginx-proxy-automation/blob/76044e46c5e448e38755059784b4cb6c9a117475/.env.sample#L68)/data/vhost.d' with the content:

```nginx
location /consumer {
        proxy_pass http://project_id.cloudfunctions.net/;
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
#        underscores_in_headers on;
#        proxy_set_header proxy_ssl_server_name $host;
}

location / {
        proxy_pass http://someurl.com;
}
```

> This will replace your whole location for the domain someurl.com and you can manage as you want. This part I have tested and it works smothly

Hope it helps!

**Reply by Borillion** [reply]
Sorry, it took me a bit to get back to you. Yes, it's a similar idea to what you're saying here

Thanks for the help, I was trying to find a way to do this with environmental variables but this is just as good. 

Here https://someurl.com/consumer is a container in a public cloud service ( In this case lets pretend AWS ) https://project_id.cloudfunctions.net is a microservice running in a different one, ( in this case its in Google Cloud )
